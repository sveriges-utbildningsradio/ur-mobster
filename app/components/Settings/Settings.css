.buttonWrap {
  position: absolute;
  right: 40px;
  top: 40px;
  z-index: 2;
}

.settingsButtonWrapper {
  display: flex;
}

.settingsButton {
  height: 24px;
  margin-left: 12px;
  width: 24px;
}

.pageWrap {
  align-items: center;
  background-color: #110737;
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;
  left: 0;
  position: absolute;
  top: 0;
  width: 100vw;
  z-index: 1;
}

.innerWrap {
  width: 950px;
}

.header {
  color: #fff;
  font-family: 'Exo 2', sans-serif;
  font-size: 55px;
  font-weight: 800;
  margin-bottom: 70px;
  letter-spacing: -0.82px;
  line-height: 0.58;
}

.settingsWrap {
  border-bottom: 2px solid #caf5ff;
  margin-bottom: 80px;
}

.settingWrap {
  align-items: center;
  border-top: 2px solid #caf5ff;
  display: grid;
  grid-template-columns: 80px auto 210px 190px;
  grid-column-gap: 60px;
  max-width: 927px;
  padding: 30px 0 30px 100px;
}

.icon {
  width: 62px;
}

.settingLabel {
  border-right: 2px solid #caf5ff;
  color: #fff;
  font-family: 'Roboto', sans-serif;
  font-size: 36px;
  text-align: left;
}
