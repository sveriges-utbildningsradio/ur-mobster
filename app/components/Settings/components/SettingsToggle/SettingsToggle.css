.wrapper {
  display: flex;
  justify-content: space-around;
}

.option {
  align-items: center;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  font-family: 'Roboto', sans-serif;
  font-size: 26px;
  font-weight: 500;
}

.button {
  background: none;
  border: none;
}

.dot {
  margin-top: 20px;
  width: 14px;
}
