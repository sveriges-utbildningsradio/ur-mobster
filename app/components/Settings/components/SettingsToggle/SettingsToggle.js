import React, { useContext } from 'react'
import styled from 'styled-components'
import dot from '../../../../assets/dot-blue.png'
import { SettingsStoreContext } from '../../../../store/store'

const Wrap = styled.ul`
  display: flex;
  justify-content: space-around;
`

const Option = styled.li`
  align-items: center;
  color: ${({ active }) => (active ? '#2cdbec' : '#fff')};
  cursor: pointer;
  display: flex;
  flex-direction: column;
  font-family: 'Roboto', sans-serif;
  font-size: 26px;
  font-weight: 500;
`

const Dot = styled.img`
  margin-top: 20px;
  width: 14px;
`

type OptionType = {
  label: string,
  value: string
}

type SettingsToggleProps = {
  active: string,
  options: OptionType[]
}

const SettingsToggle = ({ active, options }: SettingsToggleProps) => {
  const { dispatch, updateLanguage } = useContext(SettingsStoreContext)

  return (
    <Wrap>
      {options.map(option => {
        const isActive = active === option.value
        return (
          <Option
            active={isActive}
            onClick={() => dispatch(updateLanguage(option.value))}
            key={option.value}
          >
            {option.label}
            {!!isActive && <Dot src={dot} />}
          </Option>
        )
      })}
    </Wrap>
  )
}

export default SettingsToggle
