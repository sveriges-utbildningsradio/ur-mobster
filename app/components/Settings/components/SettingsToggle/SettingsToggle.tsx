import React, { useContext } from 'react'
import dot from '../../../../assets/dot-blue.png'
import { SettingsStoreContext } from '../../../../store/store'
import styles from './SettingsToggle.css'
import { LanguageValue } from '../../../../types'

type OptionType = {
  label: string
  value: LanguageValue
}

type SettingsToggleProps = {
  active: string
  options: OptionType[]
}

const SettingsToggle = ({ active, options }: SettingsToggleProps) => {
  const { dispatch, updateLanguage } = useContext(SettingsStoreContext)

  return (
    <ul className={styles.wrapper}>
      {options.map(option => {
        const isActive = active === option.value

        return (
          <li className={styles.option} key={option.value}>
            <button
              className={styles.button}
              data-e2e={`settings-toggle-${option.value}`}
              onClick={() => dispatch(updateLanguage(option.value))}
              style={{ color: isActive ? '#2cdbec' : '#fff' }}
              type="button"
            >
              {option.label}
              {!!isActive && <img alt="Dot" className={styles.dot} src={dot} />}
            </button>
          </li>
        )
      })}
    </ul>
  )
}

export default SettingsToggle
