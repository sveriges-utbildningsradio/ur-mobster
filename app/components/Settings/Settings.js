// @flow
import React, { useContext, useState } from 'react'
import { FormattedMessage } from 'react-intl'
import styles from './Settings.css'
import Cog from '../../assets/cog.png'
import Close from '../../assets/close.png'
import Time from '../../assets/time.png'
import Break from '../../assets/break.png'
import Language from '../../assets/language.png'
import DefaultButton from '../Button/DefaultButton'
import SettingsToggle from './components/SettingsToggle'
import SettingsDuration from './components/SettingsDuration/SettingsDuration'
import { SettingsStoreContext } from '../../store/store'

export const LANGUAGE_OPTIONS = [
  {
    label: 'Svenska',
    value: 'sv'
  },
  {
    label: 'English',
    value: 'en'
  }
]

export const Settings = () => {
  const {
    duration,
    breakDuration,
    breakFrequency,
    language,
    updateDuration,
    updateBreakDuration,
    updateBreakFrequency
  } = useContext(SettingsStoreContext)
  const [showingSettings, toggleShowingSettings] = useState(false)

  return (
    <>
      <button
        className={styles.buttonWrap}
        onClick={() => toggleShowingSettings(!showingSettings)}
        type="button"
      >
        {showingSettings ? (
          <span className={styles.settingsButtonWrapper}>
            <FormattedMessage id="close" />
            <img
              alt="Close settings"
              className={styles.settingsButton}
              src={Close}
            />
          </span>
        ) : (
          <img
            alt="Settings page"
            className={styles.settingsButton}
            src={Cog}
          />
        )}
      </button>
      {!!showingSettings && (
        <div className={styles.pageWrap}>
          <div className={styles.innerWrap}>
            <h2 className={styles.header}>
              <FormattedMessage id="settingsHeader" />
            </h2>
            <ul className={styles.settingsWrap}>
              <li className={styles.settingWrap}>
                <img
                  alt="Duration settings"
                  className={styles.icon}
                  src={Time}
                />
                <span className={styles.settingLabel}>
                  <FormattedMessage id="time" />
                </span>
                <SettingsDuration time={duration} updaterFn={updateDuration} />
              </li>
              <li className={styles.settingWrap}>
                <img alt="Pause settings" className={styles.icon} src={Break} />
                <span className={styles.settingLabel}>
                  <FormattedMessage id="break" />
                </span>
                <FormattedMessage id="length">
                  {msg => (
                    <SettingsDuration
                      prefixLabel={`${msg} `}
                      time={breakDuration}
                      updaterFn={updateBreakDuration}
                    />
                  )}
                </FormattedMessage>

                <FormattedMessage id="interval">
                  {msg => (
                    <SettingsDuration
                      prefixLabel={`${msg} `}
                      time={breakFrequency}
                      updaterFn={updateBreakFrequency}
                    />
                  )}
                </FormattedMessage>
              </li>
              <li className={styles.settingWrap}>
                <img
                  alt="Language settings"
                  className={styles.icon}
                  src={Language}
                />
                <span className={styles.settingLabel}>
                  <FormattedMessage id="language" />
                </span>
                <SettingsToggle active={language} options={LANGUAGE_OPTIONS} />
              </li>
            </ul>
            <DefaultButton
              handleClick={() => toggleShowingSettings(false)}
              textId="doneButton"
            />
          </div>
        </div>
      )}
    </>
  )
}

export default Settings
