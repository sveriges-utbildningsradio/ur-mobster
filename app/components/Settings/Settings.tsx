import React, { useContext, useState } from 'react'
import { FormattedMessage } from 'react-intl'
import styles from './Settings.css'
import Cog from '../../assets/cog.png'
import Close from '../../assets/close.png'
import Time from '../../assets/time.png'
import Break from '../../assets/break.png'
import Language from '../../assets/language.png'
import DefaultButton from '../Button/DefaultButton'
import SettingsToggle from './components/SettingsToggle'
import SettingsDuration from './components/SettingsDuration/SettingsDuration'
import { SettingsStoreContext } from '../../store/store'

export const LANGUAGE_OPTIONS = [
  {
    label: 'Svenska',
    value: 'sv'
  },
  {
    label: 'English',
    value: 'en'
  }
]

const Settings = () => {
  const {
    duration,
    breakDuration,
    breakFrequency,
    language,
    updateDuration,
    updateBreakDuration,
    updateBreakFrequency
  } = useContext(SettingsStoreContext)
  const [showingSettings, toggleShowingSettings] = useState(false)

  return (
    <>
      <button
        className={styles.buttonWrap}
        onClick={() => toggleShowingSettings(!showingSettings)}
        data-e2e="settings-toggle-button"
        type="button"
      >
        {showingSettings ? (
          <span className={styles.settingsButtonWrapper}>
            <FormattedMessage id="close" />
            <img
              alt="Close settings"
              className={styles.settingsButton}
              src={Close}
            />
          </span>
        ) : (
          <img
            alt="Settings page"
            className={styles.settingsButton}
            src={Cog}
          />
        )}
      </button>
      {!!showingSettings && (
        <div className={styles.pageWrap}>
          <div>
            <div className={styles.innerWrap}>
              <h2 className={styles.header} data-e2e="settings-header">
                <FormattedMessage id="settingsHeader" />
              </h2>
              <ul className={styles.settingsWrap}>
                <li className={styles.settingWrap}>
                  <img
                    alt="Duration settings"
                    className={styles.icon}
                    src={Time}
                  />
                  <span className={styles.settingLabel}>
                    <FormattedMessage id="time" />
                  </span>
                  <SettingsDuration
                    dataE2EDecrease="decrease-duration"
                    dataE2EIncrease="increase-duration"
                    time={duration}
                    updaterFn={updateDuration}
                    updateByMinute
                  />
                </li>
                <li className={styles.settingWrap}>
                  <img
                    alt="Pause settings"
                    className={styles.icon}
                    src={Break}
                  />
                  <span className={styles.settingLabel}>
                    <FormattedMessage id="break" />
                  </span>
                  <FormattedMessage id="length">
                    {msg => (
                      <SettingsDuration
                        dataE2EDecrease="decrease-break-duration"
                        dataE2EIncrease="increase-break-duration"
                        prefixLabel={`${msg} `}
                        time={breakDuration}
                        updaterFn={updateBreakDuration}
                      />
                    )}
                  </FormattedMessage>

                  <FormattedMessage id="interval">
                    {msg => (
                      <SettingsDuration
                        dataE2EDecrease="decrease-break-frequency"
                        dataE2EIncrease="increase-break-frequency"
                        prefixLabel={`${msg} `}
                        time={breakFrequency}
                        updaterFn={updateBreakFrequency}
                      />
                    )}
                  </FormattedMessage>
                </li>
                <li className={styles.settingWrap}>
                  <img
                    alt="Language settings"
                    className={styles.icon}
                    src={Language}
                  />
                  <span className={styles.settingLabel}>
                    <FormattedMessage id="language" />
                  </span>
                  <SettingsToggle
                    active={language}
                    options={LANGUAGE_OPTIONS}
                  />
                </li>
              </ul>
              <DefaultButton
                handleClick={() => toggleShowingSettings(false)}
                textId="doneButton"
              />
            </div>
          </div>
          <FormattedMessage
            id="createdBy"
            values={{
              andre: <a href="https://github.com/Aelmoznino">@Aelmoznino</a>,
              agnes: <a href="https://github.com/agnesforsell">@AgnesForsell</a>
            }}
          />
        </div>
      )}
    </>
  )
}

export default Settings
