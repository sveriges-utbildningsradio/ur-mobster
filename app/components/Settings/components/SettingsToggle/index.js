import SettingsToggle from './SettingsToggle'

export default SettingsToggle
