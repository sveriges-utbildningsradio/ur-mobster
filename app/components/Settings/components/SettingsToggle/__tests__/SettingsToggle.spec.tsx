import React from 'react'
import 'jest-dom/extend-expect'
import renderWithReactIntl from '../../../../../utils/testHelpers'
import SettingsToggle from '../SettingsToggle'
import { LANGUAGE_OPTIONS } from '../../../Settings'

describe('components/Settings/SettingsToggle', () => {
  it('renders', () => {
    const { container } = renderWithReactIntl(
      <SettingsToggle active="Svenska" options={LANGUAGE_OPTIONS} />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
