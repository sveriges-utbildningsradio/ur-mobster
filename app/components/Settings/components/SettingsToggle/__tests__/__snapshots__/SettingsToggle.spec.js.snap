// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Settings/SettingsToggle renders 1`] = `
<ul
  class="sc-bdVaJa imDAUP"
>
  <li
    class="sc-bwzfXH akcDY"
  >
    Svenska
  </li>
  <li
    class="sc-bwzfXH akcDY"
  >
    English
  </li>
</ul>
`;
